### ISSUE TITLE
Calling setDate before flatpickr is ready does not format date

### ISSUE THREAD

**Issue by http-teapot** [jacobmischka/svelte-flatpickr#22]
```js
$: if (fp) fp.setDate(value, false, dateFormat);
```

This line triggers as soon as `fp` is assigned a value but it does not guarantee Flatpickr is ready. As such when I give an initial value to the component it won't format the date and look like an ISO date.

Changing the code above to the following

```js
  let ready = false;
  if (!props.onReady) {
    props.onReady = () => (ready = true);
  }

  $: if (fp && ready) {
    fp.setDate(new Date(value), false, dateFormat);
  }
```

You probably know a better way to implement this

**Comment by eriknyk** [comment]
currently I cannot set a default value for the component,
since: 
```
$: if (fp) fp.setDate(value, false, dateFormat);
```
is not working for any case for me, and also `options.defaultDate` is not working

![image](https://user-images.githubusercontent.com/205721/118320879-71277b00-b4c2-11eb-8635-269da8efe60d.png)

Any help will be appreciated.
Best Regards.

**Comment by jacobmischka** [comment]
Whoops, sorry about forgetting about this. Should be fixed in 3.1.1 I think. Let me know if not!
